accio-main/src/main/java/io/accio/main/web/DuckDBResource.java

```diff
@@ -0,0 +1,152 @@
+            }
+            catch (AccioException e) {
+                // Sending DDL via executeQuery() still work. Should catch exception to make sense.
+                if (e.getMessage().contains("executeQuery() can only be used with queries that return a ResultSet")) {
+                    return new QueryResultDto(ImmutableList.of(), ImmutableList.of());
+                }
+                throw e;
+            }
+            catch (Exception e) {
+                throw new AccioException(GENERIC_USER_ERROR, e);
+            }
+        }).whenComplete(bindAsyncResponse(asyncResponse));
+    }
+
+    @GET
+    @Path("/settings/init-sql")
+    public void getInitSQL(@Suspended AsyncResponse asyncResponse)
+    {
+        supplyAsync(metadata::getInitSQL)
+                .whenComplete(bindAsyncResponse(asyncResponse));
+    }
+
+    @PUT
+    @Path("/settings/init-sql")
+    public void setInitSQL(
+            String sql,
+            @Suspended AsyncResponse asyncResponse)
+    {
+        runAsync(() -> {
+            metadata.setInitSQL(sql);
+            metadata.reload();
+            java.nio.file.Path initSQLPath = metadata.getInitSQLPath();
+            FileUtil.archiveFile(initSQLPath);
+            FileUtil.createFile(initSQLPath, sql);
+        }).whenComplete(bindAsyncResponse(asyncResponse));
+    }
+
+    @PATCH
+    @Path("/settings/init-sql")
+    public void appendToInitSQL(
+            String sql,
+            @Suspended AsyncResponse asyncResponse)
+    {
+        runAsync(() -> {
+            metadata.directDDL(sql);
+            metadata.setInitSQL(metadata.getInitSQL() + "\n" + sql);
+            FileUtil.appendToFile(metadata.getInitSQLPath(), sql);
+        }).whenComplete(bindAsyncResponse(asyncResponse));
+    }
+
+    @GET
+    @Path("/settings/session-sql")
+    public void getSessionSQL(@Suspended AsyncResponse asyncResponse)
+    {
+        supplyAsync(metadata::getSessionSQL)
+                .whenComplete(bindAsyncResponse(asyncResponse));
+    }
+
+    @PUT
+    @Path("/settings/session-sql")
+    public void setSessionSQL(
+            String sql,
+            @Suspended AsyncResponse asyncResponse)
+    {
+        runAsync(() -> {
+            metadata.setSessionSQL(sql);
+            metadata.reload();
+            java.nio.file.Path sessionSQLPath = metadata.getSessionSQLPath();
+            FileUtil.archiveFile(sessionSQLPath);
+            FileUtil.createFile(sessionSQLPath, sql);
+        }).whenComplete(bindAsyncResponse(asyncResponse));
+    }
+
+    @PATCH
+    @Path("/settings/session-sql")
+    public void appendToSessionSQL(
+            String sql,
+            @Suspended AsyncResponse asyncResponse)
+    {
+        runAsync(() -> {
+            metadata.getClient().closeAndInitPool();
+            metadata.setSessionSQL(metadata.getSessionSQL() + "\n" + sql);
+            FileUtil.appendToFile(metadata.getSessionSQLPath(), sql);
+        }).whenComplete(bindAsyncResponse(asyncResponse));
```

Comment:
This API won't work. Although close and init the pool, we only append and set session SQL but we don't reload the config.